### PR TITLE
{chem}[foss/2019a,intel/2019a] GPAW v1.5.2, libvdwxc v0.4.0 w/ Python 3.7.2

### PR DESCRIPTION
**Pull request by schiotz**
(created using `eb --new-pr`)

### CODE REVIEW

File: easybuild/easyconfigs/g/GPAW/GPAW-1.5.2-intel-2019a-Python-3.7.2.eb [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    '940f9db1ffcd840e5344a531f0ecdae9ca70f1876eb4ec675e69721f19997571',  # GPAW-1.5.2-slurm-submit-fix.patch
+    # GPAW-1.5.2-fix-memory-corruption-in-revtpss.patch
+    '521e6f331d07740d6b70b35ccde1cd99c789bab5a739bc6956333c19434df837',
+    # GPAW-1.5.2-fix-memory-corruption-in-regular-tpss.patch
+    '25317f8283978825aaeee04be5e0920ee62f60072564e50e482bfa37d5430906',
+]
+
+dependencies = [
+    ('Python', '3.7.2'),
+    ('ASE', '3.17.0', versionsuffix),
```
Comment (akesandgren) [inline easybuild/easyconfigs/g/GPAW/GPAW-1.5.2-intel-2019a-Python-3.7.2.eb]: Same here, add SciPy-bundle

File: easybuild/easyconfigs/g/GPAW/GPAW-1.5.2-foss-2019a-Python-3.7.2.eb [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    '940f9db1ffcd840e5344a531f0ecdae9ca70f1876eb4ec675e69721f19997571',  # GPAW-1.5.2-slurm-submit-fix.patch
+    # GPAW-1.5.2-fix-memory-corruption-in-revtpss.patch
+    '521e6f331d07740d6b70b35ccde1cd99c789bab5a739bc6956333c19434df837',
+    # GPAW-1.5.2-fix-memory-corruption-in-regular-tpss.patch
+    '25317f8283978825aaeee04be5e0920ee62f60072564e50e482bfa37d5430906',
+]
+
+dependencies = [
+    ('Python', '3.7.2'),
+    ('ASE', '3.17.0', versionsuffix),
```
[the remark this answers is not in the document]
Comment (schiotz) [reply easybuild/easyconfigs/g/GPAW/GPAW-1.5.2-foss-2019a-Python-3.7.2.eb]: Indeed!  Thank you.

File: easybuild/easyconfigs/g/GPAW/GPAW-1.5.2-intel-2019a-Python-3.7.2.eb (hunk 1 above)
Comment (schiotz) [reply easybuild/easyconfigs/g/GPAW/GPAW-1.5.2-intel-2019a-Python-3.7.2.eb]: Yes

File: easybuild/easyconfigs/g/GPAW/GPAW-1.5.2-intel-2019a-Python-3.7.2.eb [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    ('libxc', '4.3.4'),
+    ('GPAW-setups', '0.9.20000', '', True),
```
[the remark this answers is not in the document]
Comment (schiotz) [reply easybuild/easyconfigs/g/GPAW/GPAW-1.5.2-intel-2019a-Python-3.7.2.eb]: This is actually missing on purpose.  libvdwxc does not work with the Intel Math Kernel Library.  The problem is that MKL implements most of FFTWs interface but not all of it, and libvdwxc uses one of the unsupported ones.  You cannot have both FFTW and MKL linked at the same time, it core dumps apparently due to name collisions.

We hope to get this fixed in future versions of GPAW.

File: easybuild/easyconfigs/l/libvdwxc/libvdwxc-0.4.0-foss-2019a.eb [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+of density functional theory (DFT) codes."""
+
+toolchain = {'name': 'foss', 'version': '2019a'}
+
+source_urls = ['https://launchpad.net/libvdwxc/stable/%(version)s/+download/']
+sources = [SOURCE_TAR_GZ]
+checksums = ['3524feb5bb2be86b4688f71653502146b181e66f3f75b8bdaf23dd1ae4a56b33']
+
+dependencies = [
+    ('FFTW', '3.3.8'),
```
[the remark this answers is not in the document]
Comment (schiotz) [reply easybuild/easyconfigs/l/libvdwxc/libvdwxc-0.4.0-foss-2019a.eb]: OK, I'll remove it.

File: easybuild/easyconfigs/g/GPAW/GPAW-1.5.2-intel-2019a-Python-3.7.2.eb (hunk 3 above)
[the remark this answers is not in the document]
Comment (akesandgren) [reply easybuild/easyconfigs/g/GPAW/GPAW-1.5.2-intel-2019a-Python-3.7.2.eb]: Are they missing even in the external fftw interface to mkl that EasyBuild uses?

[the remark this answers is not in the document]
Comment (schiotz) [reply easybuild/easyconfigs/g/GPAW/GPAW-1.5.2-intel-2019a-Python-3.7.2.eb]: Yes, I think so.  It is some kind of MPI-FFT but I do not know the details, I only have it second-hand from the libvdwxc developer.  It should be possible to load FFTW and MKL together if one is very careful about the order of loading stuff, but GPAW does something "smart" that breaks it.  That will be fixed in 1.6, but it is not sure that we can get libwdvxc to work with MKL, it looks like there may be other issues.

**Comment by smoors** [comment]
Test report by @smoors
**SUCCESS**
Build succeeded for 8 out of 8 (3 easyconfigs in this PR)
nic426 - Linux centos linux 7.5.1804, Intel(R) Xeon(R) Gold 6148 CPU @ 2.40GHz, Python 2.7.5
See https://gist.github.com/dbaa0cea163dbfb4015ab5f3f0674f89 for a full test report.

**Comment by smoors** [comment]
Going in, thanks @schiotz!

**Comment by smoors** [comment]
not yet, needs approval from @akesandgren first...

File: easybuild/easyconfigs/g/GPAW/GPAW-1.5.2-foss-2019a-Python-3.7.2.eb [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+buildcmd = 'build --customize=eb_customize.py'
+install_target = 'install --customize=eb_customize.py'
+
+download_dep_fail = True
+use_pip = False
+
+sanity_check_paths = {
+    'files': ['bin/gpaw%s' % x for x in ['', '-analyse-basis', '-basis', '-mpisim', '-plot-parallel-timings',
+                                         '-python', '-runscript', '-setup', '-upfplot']],
+    'dirs': ['lib/python%(pyshortver)s/site-packages/' + name.lower()]
```
Comment (smoors) [inline easybuild/easyconfigs/g/GPAW/GPAW-1.5.2-foss-2019a-Python-3.7.2.eb]: style nit: please rewrite as:
```
'dirs': ['lib/python%(pyshortver)s/site-packages/%(namelower)s']
```

**Comment by schiotz** [comment]
@smoors I have updated the style as you suggested.  I apologize for the delay, I was away on holiday and conference.

**Comment by schiotz** [comment]
Obsoleted by #8768 (A newer version of GPAW has been released).

**Pull request closed without merging**